openerp/addons/base/res/res_partner.py

```diff
@@ -546,6 +569,27 @@ def create(self, vals):
         # migrating to the new API
         c_type = vals.get('company_type', self._context.get('default_company_type'))
         is_company = vals.get('is_company', self._context.get('default_is_company'))
+        parent_contact = self.env['res.partner'].search([('id', '=', vals.get('parent_id'))], limit=1)
+        gravatar = vals.get('email') and self._get_gravatar_image(vals['email']) or False
+        if parent_contact:
+            #Image for Invoice
+            if vals.get('type') == 'invoice':
+                img_path = openerp.modules.get_module_resource('base', 'static/src/img', 'money.png')
+                with open(img_path, 'rb') as f:
+                    image = f.read()
+                vals['image'] = tools.image_resize_image_big(image.encode('base64'))
+            #Image for Delivery
+            if vals.get('type') == 'delivery':
+                img_path = openerp.modules.get_module_resource('base', 'static/src/img', 'truck.png')
+                with open(img_path, 'rb') as f:
+                    image = f.read()
+                vals['image'] = tools.image_resize_image_big(image.encode('base64'))
```

Comment:
You can create internal function, instead of writing code twice for loading resource
OR better
```
if vals.get('type') in ['invoice', 'delivery']:
    img_path = openerp.modules.get_module_resource('base', 'static/src/img', 'money.png' if vals.get('type') == 'invoice' else 'truck.png')
....
```